src/amo/reducers/addonsByAuthors.js

```diff
@@ -228,6 +228,33 @@ const reducer = (
           ...newState.byAddonSlug,
           [forAddonSlug]: undefined,
         };
+      } else if (authorUsernames.length) {
+        // Potentially remove add-ons loaded for these authors with this add-on
+        // type, so that we can load new add-ons in the UI (pagination). We do
+        // not do it when `forAddonSlug` is defined to avoid re-fetching the
+        // add-ons by authors on an add-on detail page.
+
+        const addonsToRemove = getAddonsForUsernames(
+          newState,
+          authorUsernames,
+          addonType,
+        );
+
+        if (addonsToRemove) {
+          for (const addonToRemove of addonsToRemove) {
+            if (addonToRemove.authors) {
+              for (const author of addonToRemove.authors) {
+                newState.byUsername[author.username] = newState.byUsername[
```

Comment:
Thanks for the explanation. I understand that `addonType` is important now. The comment up above didn't help me understand for some reason. When I re-read it, I think the comment does explain it.